crates/matrix-sdk-common/src/executor.rs

```diff
@@ -14,24 +14,28 @@
 
 //! Abstraction over an executor so we can spawn tasks under WASM the same way
 //! we do usually.
-
-#[cfg(target_arch = "wasm32")]
 use std::{
     future::Future,
     pin::Pin,
     task::{Context, Poll},
 };
 
+use futures_util::FutureExt;
 #[cfg(target_arch = "wasm32")]
 pub use futures_util::future::Aborted as JoinError;
 #[cfg(target_arch = "wasm32")]
-use futures_util::{
-    future::{AbortHandle, Abortable, RemoteHandle},
-    FutureExt,
-};
+use futures_util::future::{AbortHandle, Abortable, RemoteHandle};
 #[cfg(not(target_arch = "wasm32"))]
 pub use tokio::task::{spawn, JoinError, JoinHandle};
 
+
+/// A `Box::pin` future that is `Send` on non-wasm, and without `Send` on wasm.
+#[cfg(target_arch = "wasm32")]
+pub type BoxFuture<'a, T> = Pin<Box<dyn Future<Output = T> + 'a>>;
+#[cfg(not(target_arch = "wasm32"))]
+pub type BoxFuture<'a, T> = Pin<Box<dyn Future<Output = T> + Send + 'a>>;
```

Comment:
```suggestion
pub type BoxFuture<'a, T> = Pin<Box<dyn Future<Output = T> + SendOutsideWasm + 'a>>;
```

Reply:
I see this definition already existed though, why did you move it here?

Reply:
I needed to expose the `.boxed` method on the wasm version of the code (it was present on the real future, but not the Wasm version).  It seemed like this was where the wasm future was being defined, so I wanted to move the BoxFuture method here so it can be exposed in the same place.